pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -695,4 +698,12 @@ public void setSegmentPartitionConfig(SegmentPartitionConfig segmentPartitionCon
   public SegmentPartitionConfig getSegmentPartitionConfig() {
     return segmentPartitionConfig;
   }
+
+  public void setStarTreeIndexSpec(StarTreeIndexSpec starTreeIndexSpec) {
```

Comment:
You are right, by the time the code was implemented these methods ended up being unused. I have removed them.